backend/src/models/db.js

```diff
@@ -11,11 +11,11 @@ const DB_DATA = {
 module.exports = {
   query: async (text, params) => {
     const client = new Client(DB_DATA);
-    await client.connect();
-    return client.query(text, params).then(res => {
-      client.end();
-      return res.rows;
-    });
+    return client.connect()
+                 .catch(() => { throw new Error('DB connection error');})
+                 .then(() => client.query(text, params))
```

Comment:
Do we need to catch and the possible SQL query errors as well?

Reply:
We do, but on later stages, closer to the router. Thanks to promise nature, it will popup later in the code and will be detected by another catch block, which can return proper error message in response.